src/js/components/Distribution/Distribution.js

```diff
@@ -15,107 +15,104 @@ Value.propTypes = {
+        return true;
+      }
+      return false;
+    });
+
+    if (subIndex === values.length) {
       const value = values[0];
       return (
         <Value value={value} basis={basis}>
           {children(value)}
         </Value>
       );
     }
-    if (values.length > 1) {
-      // calculate total
-      let total = 0;
-      values.forEach(v => {
-        total += v.value;
-      });
-
-      // figure out how many of the values area needed to represent half of the
-      // total
-      let subTotal = 0;
-      let subIndex;
-      values.some((v, index) => {
-        subTotal += v.value;
-        if (subTotal >= total * 0.4) {
-          subIndex = index + 1;
-          return true;
-        }
-        return false;
-      });
 
-      if (subIndex === values.length) {
-        const value = values[0];
-        return (
-          <Value value={value} basis={basis}>
-            {children(value)}
-          </Value>
-        );
-      }
-
-      let childBasis;
-      if (subTotal > total * 0.7) {
-        childBasis = ['3/4', '1/4'];
-      } else if (subTotal > total * 0.6) {
-        childBasis = ['2/3', '1/3'];
-      } else {
-        childBasis = ['1/2', '1/2'];
-      }
+    let childBasis;
+    if (subTotal > total * 0.7) {
+      childBasis = ['3/4', '1/4'];
+    } else if (subTotal > total * 0.6) {
+      childBasis = ['2/3', '1/3'];
+    } else {
+      childBasis = ['1/2', '1/2'];
+    }
 
-      return (
-        <Box
-          direction={direction}
-          basis={basis}
-          flex={basis ? 'shrink' : true}
-          overflow="hidden"
+    return (
+      <Box
+        direction={direction}
+        basis={basis}
+        flex={basis ? 'shrink' : true}
+        overflow="hidden"
+        gap={gap}
+        fill={fill}
+        {...rest}
+      >
+        <Distribution
+          values={values.slice(0, subIndex)}
+          basis={childBasis[0]}
+          direction={direction === 'row' ? 'column' : 'row'}
           gap={gap}
-          fill={fill}
-          {...rest}
         >
-          <Distribution
-            values={values.slice(0, subIndex)}
-            basis={childBasis[0]}
-            direction={direction === 'row' ? 'column' : 'row'}
-            gap={gap}
-          >
-            {children}
-          </Distribution>
-          <Distribution
-            values={values.slice(subIndex)}
-            basis={childBasis[1]}
-            direction={direction === 'row' ? 'column' : 'row'}
-            gap={gap}
-          >
-            {children}
-          </Distribution>
-        </Box>
-      );
-    }
-    return null;
+          {children}
+        </Distribution>
+        <Distribution
+          values={values.slice(subIndex)}
+          basis={childBasis[1]}
+          direction={direction === 'row' ? 'column' : 'row'}
+          gap={gap}
+        >
+          {children}
+        </Distribution>
+      </Box>
+    );
   }
-}
+  return null;
+};
+
+Distribution.defaultProps = {
```

Comment:
Normally I'll suggest assigning these values as part of the props deconstruction and avoid the usage of `defaultProps`, but the content looks a little too busy to be assigned there (lines 19-26). I'm approving this PR, but I wanted to see if anyone else has other thoughts here.

Reply:
I'm answering myself here, but I do think this should be moved up to the component declaration and used while deconstructing the props. thanks!

Reply:
@ShimiSun  did the required changes, But not sure why readme.tests are failing?